warehouse/integrations/deltalake-native/deltalake_native.go

```
@@ -0,0 +1,1367 @@
+		}
+	}
+
+	var (
+		affected int64
+		updated  int64
+		deleted  int64
+		inserted int64
+	)
+
+	if d.LoadTableStrategy == appendMode {
+		err = row.Scan(&affected, &inserted)
+	} else {
+		err = row.Scan(&affected, &updated, &deleted, &inserted)
+	}
+
+	if err != nil {
+		return map[string]error{
+			warehouseutils.IdentifiesTable: nil,
+			warehouseutils.UsersTable:      fmt.Errorf("getting rows affected for dedup: %w", err),
+		}
+	}
+
+	d.Stats.NewTaggedStat("dedup_rows", stats.CountType, stats.Tags{
+		"sourceID":    d.Warehouse.Source.ID,
+		"sourceType":  d.Warehouse.Source.SourceDefinition.Name,
+		"destID":      d.Warehouse.Destination.ID,
+		"destType":    d.Warehouse.Destination.DestinationDefinition.Name,
+		"workspaceId": d.Warehouse.WorkspaceID,
+		"namespace":   d.Namespace,
+		"tableName":   warehouseutils.UsersTable,
+	}).Count(int(updated))
+
+	d.Logger.Infow("completed loading for users and identifies tables",
+		logfield.SourceID, d.Warehouse.Source.ID,
+		logfield.SourceType, d.Warehouse.Source.SourceDefinition.Name,
+		logfield.DestinationID, d.Warehouse.Destination.ID,
+		logfield.DestinationType, d.Warehouse.Destination.DestinationDefinition.Name,
+		logfield.WorkspaceID, d.Warehouse.WorkspaceID,
+		logfield.Namespace, d.Namespace,
+	)
+
+	return map[string]error{
+		warehouseutils.IdentifiesTable: nil,
+		warehouseutils.UsersTable:      nil,
+	}
+}
+
+func (d *Deltalake) tableLocationQuery(tableName string) string {
+	var (
+		enableExternalLocation = warehouseutils.GetConfigValueBoolString(EnableExternalLocation, d.Warehouse)
+		externalLocation       = warehouseutils.GetConfigValue(ExternalLocation, d.Warehouse)
+	)
+
+	if enableExternalLocation != "true" {
+		return ""
+	}
+	if externalLocation == "" {
+		return ""
+	}
+	return fmt.Sprintf("LOCATION '%s/%s/%s'", externalLocation, d.Namespace, tableName)
+}
+
+func (d *Deltalake) dropDanglingStagingTables() {
+	tableNames, _ := d.fetchTables("^rudder_staging_.*$")
```

isn't dangerous to ignore error here. Since, there is a possibility that there is nothing in `tableNames` because some error occured while fetching the tables names. This will led to the accumulation of those tables. I believe we should atleast log it.

---

Good point. let me add error logs here.

---

Although, we are dropping staging tables after every sync. But there might be a case where we didn't have DROP permissions and since we are skipping these errors, these tables will accumulate.

---

Added warning logs for this.